src/js/serial_backend.js

```diff
@@ -301,6 +316,23 @@ function onOpen(openInfo) {
     }
 }
 
+function onOpenVirtual() {
+    GUI.connected_to = GUI.connecting_to;
+    GUI.connecting_to = false;
+
+    CONFIGURATOR.connectionValid = true;
+
+    mspHelper = new MspHelper();
```

Comment:
Maybe also implement a `finishCloseVirtual()`  to:

- call MSP.clearListener()
- Reset some CONFIGURATOR parameters like connectionValid, cliValid, cliActive
- and finally call MSP.disconnect_cleanup()

See `finishClose` and `onClosed` in serial_backend.js or refactor these functions.

Reply:
I don't think this is necessary, I believe the current `finishClose` and `onClosed` take care of everything.